### PR TITLE
Release lock before program exits

### PR DESCRIPTION
**Pull request by jenny-cheung**
A fix patch for issue#9324 to avoid the unreleased lock.

### CODE REVIEW

File: src/postgres/src/test/thread/thread_test.c [hunk 1]
```diff
@@ -157,6 +158,7 @@ main(int argc, char *argv[])
 		 * 1 that uses it, so avoid using it.
 		 */
 		fprintf(stderr, "Failed to create thread 2 **\nexiting\n");
+        pthread_mutex_unlock(&init_mutex);
```
Comment (tedyu) [inline src/postgres/src/test/thread/thread_test.c:161]: Please align pthread_mutex_unlock() with the rest of the code (left indent 2 chars).

**Comment by tedyu** [comment]
Please sign CLA.

After indentation is fixed, I will run your patch through test suite.

**Comment by jenny-cheung** [comment]
The indentation has been fixed.

**Review by d-uspenskiy — COMMENTED** [review]
Hello @jenny-cheung could you please describe the impact of the issue? What is going wrong without your fix?
As you know YugaByte uses postgres fork and this code belong to postgres native code. Latest version of postgres sources has the same code now. There is no mutex unlocking before exit https://github.com/postgres/postgres/blob/5865e064abfbbe11ebfc09881be009c0f69b4dc2/config/thread_test.c#L129

In case it is real problem it will be good to create issue in postgres project as well.

**Comment by jenny-cheung** [comment]
@d-uspenskiy Hi, the impact is that the lock is not correctly released before exit. It is not a very harmful bug( e.g., resource leak, deadlock) since the program exits and all of the program resources will be cleaned up by some common OS. However, other OS systems (e.g., embedded systems) do not automatically free some resources at the exit. It is a good manner for resource management. Adding the unlock statement adds symmetry, so the code looks better. Also, the debugger would not warn this case : )

**Comment by d-uspenskiy** [comment]
Issue was closed by the originator. No fix planned.

**Pull request closed without merging**